### ISSUE TITLE
[BUG] Correct train-validation-split for the cross validation

### ISSUE THREAD

**Issue by franneck94** [franneck94/TensorCross#32]
# System information

OS Platform and Distribution: All
TensorFlow version: 2.3
Python version: 3.8

# Source code / logs

Correct the train-validation split of the BaseSearchCV's fit method (see [here](../blob/main/tensorcross/model_selection/search_cv.py)).
The validation set should be "slided" over the passed in dataset, such that every data point was once in the validation set.

Be sure to write tests for this new feature.

**Comment by franneck94** [comment]
Fixed issue in a41b379
